code/datums/components/bump_attack.dm

```diff
@@ -20,6 +20,7 @@
 	toggle_action.give_action(parent)
 	toggle_action.update_button_icon(active)
 	RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, toggle_path)
+	RegisterSignal(parent, COMSIG_MOVABLE_BUMP, bump_action_path)
```

Comment:
```suggestion
	if(active)
		RegisterSignal(parent, COMSIG_MOVABLE_BUMP, bump_action_path)
```

Reply:
I'm not too sure what problem this solves - that line of code is in /datum/component/bump_attack/Initialize(). I don't actually know when it's called, but I think it's only called once per mob - and active is hard-coded = true at the beginning of the component.

Is this to make it easier to alter bump attack to FALSE in the future? I suppose that would be reasonable.

Reply:
ideally you should be passing active as a parameter to the component initialize, then xenos adding the component pass active from some config value rather than hard coding it